### PR TITLE
fix datetimes in `Group` and `GroupList` `to_pandas`

### PR DESCRIPTION
**Pull request by haakonvt**
## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/iam.py [hunk 1]
```diff
@@ -51,24 +54,58 @@ def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> S
         return cls(
             name=resource["name"],
             source_id=resource.get("sourceId"),
-            capabilities=[Capability.load(c) for c in resource.get("capabilities", [])],
-            id=resource.get("id"),
-            is_deleted=resource.get("isDeleted"),
+            capabilities=[Capability.load(c) for c in resource.get("capabilities", [])] or None,
+            id=resource["id"],
+            is_deleted=resource["isDeleted"],
```
Comment (doctrino) [inline cognite/client/data_classes/iam.py:59]: This class is used for both read and write, and `id` and `isDeleted` are not required for the `write` version. So I think they need to be optional.

File: cognite/client/data_classes/iam.py [hunk 2]
```diff
@@ -51,24 +54,58 @@ def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> S
...
+        camel_case: bool = False,
+        convert_timestamps: bool = True,
+    ) -> pd.DataFrame:
+        df = super().to_pandas(expand_metadata, metadata_prefix, ignore, camel_case, convert_timestamps)
+
+        # The API uses -1 to represent "no deleted time". It looks weird if deleted = False,
+        # but deleted_time = 1969-12-31 23:59:59.999:
+        key = "deletedTime" if camel_case else "deleted_time"
+        if self.deleted_time == -1 and convert_timestamps and key in df.index:
+            df.at[key, "value"] = local_import("pandas").NaT
```
Comment (doctrino) [inline cognite/client/data_classes/iam.py:85]: Nice

File: docs/source/settings.rst [hunk 3]
```diff
@@ -23,7 +23,7 @@ You can set global configuration options like this:
     global_config.max_retries = 10
     global_config.max_retry_backoff = 10
     global_config.max_connection_pool_size = 10
-    global_config.status_forcelist = {429, 502, 503}
+    global_config.status_forcelist = {429, 502, 503, 504}
```
Comment (doctrino) [inline docs/source/settings.rst:26]: Why this one?

**Review by doctrino — COMMENTED** [review]
I think the id and isDeleted need to stay optional, also when you load them, as you may what to load a group from disk.

File: cognite/client/data_classes/iam.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/data_classes/iam.py:59]: Oh right... yeah guess so; will revert. Would be awkward to force people to define it as `id: None, isDeleted: None` in the config :(

File: docs/source/settings.rst (hunk 3 above)
Comment (haakonvt) [reply docs/source/settings.rst:26]: Updating docs to match actual `status_forcelist`

**Pull request merged**